src/test/codegen/c-variadic-opt.rs

```diff
@@ -14,6 +14,17 @@ extern "C" {
 #[no_mangle]
 pub unsafe extern "C" fn c_variadic_no_use(fmt: *const i8, mut ap: ...) -> i32 {
     // CHECK: call void @llvm.va_start
-    vprintf(fmt, ap)
+    vprintf(fmt, ap.as_va_list())
+    // CHECK: call void @llvm.va_end
+}
+
+// Check that `VaListImpl::clone` gets inlined into a direct call to `llvm.va_copy`
+#[no_mangle]
+pub unsafe extern "C" fn c_variadic_va_copy(fmt: *const i8, mut ap: ...) -> i32 {
+    let mut ap2 = ap.clone();
```

Comment:
It is tested elsewhere, but can you add a `CHECK: call void @llvm.va_start` here for the call to `va_start` on `ap`.